Release notes for 8.8.0 (#15038)
elastic/logstash: docs/static/releasenotes.asciidoc

* <<logstash-8-0-0-alpha1,Logstash 8.0.0-alpha1>>

[[logstash-8-8-0]]
=== Logstash 8.8.0 Release Notes

[[features-8.8.0]]
==== New features and enhancements

> * Negative acknowledgment of a batch. With this functionality a plugin can request the abortion a batch of event.
>   When used in combination with persisted queue it enables the reprocessing on Logstash restart https://github.com/elastic/logstash/pull/14940[#14940]

Reviewer: ```suggestion
* Adds functionality for negative acknowledgment when a batch contains errors or was not received.   
With this new functionality, a plugin can request that processing be stopped when a problem is detected in a batch.
When used in combination with a persistent queue, the batch is eligible for reprocessing when {ls} restarts.https://github.com/elastic/logstash/pull/14940[#14940]
```

Reviewer: Above is a suggestion for rewording.  The wording suggestions are optional except for this: The word "abortion" must be replaced.

Reviewer 2: Hi @karenzone I always appreciate a lot of your suggestion, and in particular this. 
Me neither like that word, but nothing more nice comes to me. 
The concept I want to express is: when a plugin rises that error not only it's asking to stop processing but also to repeat the processing of the same batch next it run. 
A stop on of a processing is "stop doing what you are doing and drop the data" while this kind of request is "stop what you are doing and permit (if the queue agrees on this) to reprocess the full batch from start (of the filters section)".

* <<logstash-8-0-0-alpha1,Logstash 8.0.0-alpha1>>

[[logstash-8-8-0]]
=== Logstash 8.8.0 Release Notes

[[features-8.8.0]]
==== New features and enhancements

* Negative acknowledgment of a batch. With this functionality a plugin can request the abortion a batch of event.
  When used in combination with persisted queue it enables the reprocessing on Logstash restart https://github.com/elastic/logstash/pull/14940[#14940]

> [[dependencies-8.9.0]]

Reviewer 3: ```suggestion
[[dependencies-8.8.0]]
```

==== Updates to dependencies

* Updates Bundler to version 2.4 https://github.com/elastic/logstash/pull/14995[#14995]

==== Plugins

*Elasticsearch Filter - 3.15.0*

* Standardize SSL settings to comply with Logstash's naming convention https://github.com/logstash-plugins/logstash-filter-elasticsearch/pull/168[#168]

* Added support for configurable retries with new `retry_on_failure` and `retry_on_status` options https://github.com/logstash-plugins/logstash-filter-elasticsearch/pull/160[#160]

*Memcached Filter - 1.2.0*

* Upgrade Dalli to 3.x https://github.com/logstash-plugins/logstash-filter-memcached/pull/33[#33]

*Beats Input - 6.6.0*

* Standardize SSL settings to comply with Logstash's naming convention https://github.com/logstash-plugins/logstash-input-beats/pull/470[#470]

*Elasticsearch Input - 4.17.0*

* Standardize SSL settings to comply with Logstash's naming convention https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/185[#185]

*Http Input - 3.7.0*

* Standardize SSL settings to comply with Logstash's naming convention https://github.com/logstash-plugins/logstash-input-http/pull/165[#165]

*Kafka Integration - 11.2.1*

> * Fix nil exception to empty headers of record during event metadata assignment https://github.com/logstash-plugins/logstash-integration-kafka/pull/140[#140]
> 
> * Added TLS truststore and keystore settings specifically to access the schema registry https://github.com/logstash-plugins/logstash-integration-kafka/pull/137[#137]
> 
> * Added config `group_instance_id` to use the Kafka's consumer static membership feature https://github.com/logstash-plugins/logstash-integration-kafka/pull/135[#135]
> 
> * Changed Kafka client to 3.3.1, requires Logstash >= 8.3.0. 
> * Deprecated `default` value for setting `client_dns_lookup` forcing to `use_all_dns_ips` when explicitly used https://github.com/logstash-plugins/logstash-integration-kafka/pull/130[#130]
> * Changed the consumer's poll from using the one that blocks on metadata retrieval to the one that doesn't https://github.com/logstash-plugins/logstash-integration-kafka/pull/133[#136]

Reviewer 3: ```suggestion
* Fix nil exception to empty headers of record during event metadata assignment https://github.com/logstash-plugins/logstash-integration-kafka/pull/140[#140]
* Added TLS truststore and keystore settings specifically to access the schema registry https://github.com/logstash-plugins/logstash-integration-kafka/pull/137[#137]
* Added config `group_instance_id` to use the Kafka's consumer static membership feature https://github.com/logstash-plugins/logstash-integration-kafka/pull/135[#135]
* Changed Kafka client to 3.3.1, requires Logstash >= 8.3.0. 
* Deprecated `default` value for setting `client_dns_lookup` forcing to `use_all_dns_ips` when explicitly used https://github.com/logstash-plugins/logstash-integration-kafka/pull/130[#130]
* Changed the consumer's poll from using the one that blocks on metadata retrieval to the one that doesn't https://github.com/logstash-plugins/logstash-integration-kafka/pull/133[#136]
```

*Normalize_config_support Mixin - 1.0.0*

*Elasticsearch Output - 11.15.1*

> * Fixed race condition during plugin registration phase https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1125[#1125]
> 
> * Added the ability to negatively acknowledge the batch under processing if the plugin is blocked in a retry-error-loop and a shutdown is requested. https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1119[#1119]
> 
> * Standardize SSL settings to comply with Logstash's naming convention https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1118[#1118]

Reviewer 3: ```suggestion
* Fixed race condition during plugin registration phase https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1125[#1125]
* Added the ability to negatively acknowledge the batch under processing if the plugin is blocked in a retry-error-loop and a shutdown is requested. https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1119[#1119]
* Standardize SSL settings to comply with Logstash's naming convention https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1118[#1118]
```

* <<logstash-8-0-0-alpha1,Logstash 8.0.0-alpha1>>

[[logstash-8-8-0]]
=== Logstash 8.8.0 Release Notes

[[notable-8.8.0]]
==== Notable issues fixed

* In previous Logstash versions, when a pipeline used Elasticsearch output and the credentials expired, it ended in the
  pipeline remained stuck in disconnecting from an invalidated Elasticsearch endpoint also when the pipeline definition
  was updated with the new connection credentials.
  The issue https://github.com/elastic/logstash/issues/14739[14739] describe how to reach such condition in terms of Central Pipeline Management.
  To fix this problem, Logstash core has been modified so that a plugin can request a termination of the inflight batch
  during the shutdown of itself.
  The bugfix https://github.com/elastic/logstash/pull/14940[#14940] solves the original issue,
  avoiding any data loss when used in combination with persistent queues.

> 

Reviewer 4: Maybe something like:

Fix a race condition that prevents Logstash from updating a pipeline's configuration with in-flight events experiencing connection errors. https://github.com/elastic/logstash/issues/14739[14739]

This issue primarily manifests following the update of Elasticsearch credentials through Central Management, after they expired while events were in-flight. It causes the Elasticsearch Output to get stuck attempting to send events with the expired credentials instead of using the updated ones.

To address this problem, Logstash has improved the pipeline shutdown phase, to allow the output plugin to request the termination of the in-flight batch of events, preventing the need for administrators to manually restart Logstash. Furthermore, when used in combination with a persistent queue to avoid data loss, the batch is eligible for reprocessing on pipeline restart.

Plugin developers can also decide to make use of such functionality on other output plugins.

> 

Reviewer 2: Fuxed with d8225a431f99deb0545fceb70aa13051a69505ba
